core-plugin/src/com/google/gct/idea/debugger/ProjectRepositoryValidator.java

```diff
@@ -123,19 +145,36 @@ public SyncResult checkSyncStashState() {
     }
 
     if (!foundDebuggee) {
-      return new SyncResult(/*isinvalid*/true,
+      return new SyncResult(/*isinvalid*/ true,
                             /*needsstash*/ false,
                             /*needssync*/ false,
-                            /*isdeterminable*/ true,
                             /*target SHA*/ null,
-                            /*target repo*/ null);
+                            /*target repo*/ null,
+                            /* hasCloudRepository */ false,
+                            /* repoType */ "Unknown");
     }
 
     GitRepository targetLocalRepo = null;
+    String revisionId = null;
+
+    // shouldn't be more than one repo but if there is, we pick cloud repos
     if (cloudRepo != null) {
+      revisionId = cloudRepo.getRevisionId();
+      repoType = GctBundle.getString("clouddebug.cloud.repository");
+    }
+    else if (gerritRepo != null) {
+      revisionId = gerritRepo.getRevisionId();
+      repoType = GctBundle.getString("clouddebug.gerrit");
+    }
+    else if (otherGitRepo != null) {
+      revisionId = otherGitRepo.getRevisionId();
+      repoType = GctBundle.getString("clouddebug.nongoogle.git");
```

Comment:
does this have to be highlighted as 'non-google' and 'cloud'?  Really this seems like the default git case.